robottelo/ui/contentviews.py

```diff
@@ -646,6 +646,50 @@ def package_search(self, name, version_name, package_name,
         return self.wait_until_element(
             locators['contentviews.version.package_name'] % package_name)
 
+    def fetch_version_packages(self, name, version_name):
+        """Return a list of all the packages inside specific content view
+        version"""
+        self.click(self.version_search(name, version_name))
+        self.click(tab_locators['contentviews.tab_version_packages'])
+        packages = []
+        while True:
```

Comment:
try this:
```python
next_ = True
while next_:
    # do what you want here
    ...
     next_ = self.find_element(
                   locators['contentviews.version.content_next_page'])
     if next_:
         self.click(next_)
```

Reply:
And what's the core difference? How will it help to handle your early mentioned case when we have next button 'broken' and always available?
Imo it will end up in a similar flow with the same drawbacks, the only thing which really differs - few extra checks which decrease readability.

Reply:
@abalakh I do not agree with you, please look at code, 

will ack and will not block this PR , let other other reviewers decide.

Reply:
I'm not against such approach, i just can't see real benefits in it. @rochacbruno @renzon what'd you suggest? I mean the best pythonic equivalent of `do .. while` or `repeat .. until`

Reply:
@abalakh @ldjebran @renzon 

I am against using an endless loop as I pointed in https://github.com/SatelliteQE/robottelo/issues/4190 

For this case I think we should use a `retry` with a limited counter

```python
with retry(times=30, sleep_interval=..., exceptions=...):
    # do things
    next_ = self.find_element(
                   locators['contentviews.version.content_next_page'])
    self.click(next_)  # if this operation fails and raise an error the operation will be retried for more 30 times
```
as you can see in #4190 we need to choose an external package that implements this (there are many of them)

If not using the above approach, I prefer to use @abalakh  version using the `while` with a `break` I think it is more readable.

```python
while True:
    if cond:
        break
```

is the Pythonic version of `repeat....until`

But I would prefer to remove all `while True` from our code and start using a `retry` context.

Reply:
@rochacbruno even if it's known that this is the replacement of repeat....until, even if I know that this is a working scenario. I will never recommend it's usage (this is a very bad programming habit let it without a condition).

Reply:
@ldjebran When I was a Java programmer that community really avoid such king of loop. But there since there we have the mentioned construction (do while), code readability isn't compromised most of times.

Once I started to code in Python I realized people are ok with infinite loop with some break condition inside it to improve readbility, not having to create a flag only to control flux.

So in this case I agree with @rochacbruno. Using a a retry function would be the best, otherwise I would go with while True: break.

Reply:
@renzon we have to write to Guido to change that and implement do while :)

Reply:
In the meantime - we have a separate issue to decide which `retry` implementation is the best for us, this place is yet another one which should be replaced in scope of that issue. Let's not block this PR which addresses different issue and uses preferred approach (3 votes for it and 1 against 🙂 )